### ISSUE TITLE
Reminders - Permissions Issue

### ISSUE THREAD

**Issue by Karakth** [greghesp/assistant-relay#245]
Attempting to set a reminder using Assistant Relay results in the response, "Sure, but you have to enable personal results on this device" or something to that effect. On my assistant settings, I do not have any option to enable personal results on the API.

**Comment by Daniele70** [comment]
Hello.
Not a solution but some elements I also posted on another tread  about this question of Personal Results :
    
First of all, many thanks to Greg and all other contributors : this project is very usefull !
Now my problem … may-be someone can give me some suggestions.
I installed assistant realy to be able to interface openhab to devices not supported in openhab but compatible with Google.
In particular a door lock that I’d like to close thru openhab.
Since my competencies are limited, I really followed all the instructions and now everything is running fine. In the sandbox I can send commands to google assistant and broadcast messages to selected google home mini.
Unfortunately there is something not yet working : when I try to close my door or simply I ask : « Who am I ? » I get as answer that I need to enable personal results access.
I immediatly get on my phone a pop-up request to grant personal results, but the request is from an empty device name. In this case if I try to activate personal results thru the slider, the app crashes.
If I try to fill a name it crashes also or do not register so no progress.
Tryed on several android devices but always same result (so not a problem of the android app)
NB : the SDK do not appear as a device in Google Home (not sure this will make any difference)
I tryed to connect assistant realy with « TV and limited input device » handly created or with credentials coming from the device created in the project but no difference.
I’m struggling to understand if the problem is coming from assistant relay not transferring in some way the device identification or from a problem on google API.
But it seems some peoples could make it work and this shouldn’t be a limitation of the API since.
I’m wondering if I missed something in terms of further device registration after the automatic procedure going thru the registration tool but I’m really at the limits of my skills 😊

Any suggestions is welcome !
Thanks !

PS : I could progress removing everything and registering the device directly from command line thru registerDevice.js. After that the device appear in Google Home, in the Action console under my project and also as credentials in the API ... but always same result. It seems the request arrive with no link to the device created thru registerDevice.js

**Comment by Karakth** [comment]
What's weird is that this was working up until a couple of months ago.

**Comment by Daniele70** [comment]
Thanks for the comment.
When I'll have more time, even if I have a limited understanding of what I'm doing, I'll make some additional tests to try to figure out if this is coming from a change / bug in the google API or from a problem in the local config of assistant relay.
If someone could confirm it's still working for him or not, this could help me to look in the right direction.
Thanks !

**Comment by florian-mollin** [comment]
Hi,

> I immediatly get on my phone a pop-up request to grant personal results, but the request is from an empty device name. In this case if I try to activate personal results thru the slider, the app crashes.

This is exactly the same problem for me !

Assitant Relay worked fine for a few months, and then impossible to get it to work (personal results)...

I'm pretty sure it's permissions issues, but how to properly "enable personal results" ?

**Comment by Karakth** [comment]
I think a possible solution might be here:

https://developers.google.com/assistant/sdk/reference/device-registration/register-device-manual

The problem seems to be that Assistant is asking to set permissions on the "device" but the device is Null.

**Comment by Daniele70** [comment]
Hello,
yes, I yet tried with the registerDevice.js file provided within the installation of assistant relay and that I suppose is intedend to do exactly what is described in the procedure you mentionned.
I could than get a new device in Google Home so the procedure was probably fine, but still same result : a request to autorize personal results from a Null device on my phone.
I can just imagine that something changed on google side in the way the requesting device is identified and now this is incompatible with the way Assitant Realy send the requests. But I'm not really familiar with all this stuff :-)

**Comment by willthong** [comment]
I have the same issue with the following Sandbox parameters:

```
"command": "tell me my agenda",
"converse": false,
"broadcast": false
```

Response: "First I'll need your permission to give personal results. You can make that change in Google Assistant on your phone."

On my phone, the Google app gives a notification entitled "Allow personal answers". When I click the 'Personal results' toggle to turn it on, the app crashes (presumably because the device name is blank). When I set the device name to "Assistant Relay" or "Null", the app doesn't crash but the problem is also not resolved.

**Comment by wallnerhaas** [comment]
Was anyone able to resolve this - I'm in the same boat that personalized results used to work and now they don't work any longer. I get the same message on my phone to allow personalized results for a device without name.

**Comment by Karakth** [comment]
Nope, afraid I have no solution on my end. My personal workaround was to
use a bit of python script in appdaemon to replace the Google Assistant
reminders with Home Assistant app notifications.

**Comment by Daniele70** [comment]
Nope on my side also.
In reality I gave up the researches since you can still utilize all the commands that do not require user identification and I found a bypass for the only one that was usefull to me (open / close a door locker from TheKeys thru openhab)

**Comment by thematrixdev** [comment]
I have just found Assistant-Relay and using it. Hopefully it is not too late to answer. I had the same problem. To solve this, POST an command which requires personal-result, such as "set a reminder". There will be a new notification asking for personal-result on your Android phone. Click on the notification to open it. Give the device a name, and then turn on personal result. Clicking on "approve", or turning-on without giving a device name will crash the app.

**Comment by Karakth** [comment]
> I have just found Assistant-Relay and using it. Hopefully it is not too late to answer. I had the same problem. To solve this, POST an command which requires personal-result, such as "set a reminder". There will be a new notification asking for personal-result on your Android phone. Click on the notification to open it. Give the device a name, and then turn on personal result. Clicking on "approve", or turning-on without giving a device name will crash the app.

This didn't work for me. Although I added a device name and allowed permissions, POSTing another command which requires a personal result still results in the same no-name notification.

**Comment by amosyuen** [comment]
Same problem adding a device name doesn't resolve it

**Comment by cthulu** [comment]
I was facing the same problem on my Android phone - could never confirm the notification. However on my iPhone I've installed the Assistant app and when I got the notification there I confirmed it and since then I can access "personal data" via Assistant Relay ¯\_(ツ)_/¯

I guess the iOS has `null` device name better addressed?

**Comment by ryancasler** [comment]
You're saying that you are able to do something that normally requires voice recognition, like setting a reminder, from Google Assistant Relay?   When you say you were never able to "confirm the notification", what exactly do you mean?   
Because there is no device to enable personal results for when using Google Assistant Relay.   So, that function should not be available.  I tried adding my account to an iPhone's Google Assistant and was still not able to do anything that required personal results.

**Comment by cthulu** [comment]
I was able to add things to my personal shopping list for example.
It wasn't possible to confirm the notification on my android phone but
after clicking it on my iPhone it started working. Not sure why or how 🤷‍♂
